File: NewsApi/Models/Category.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace NewsApi.Models
{
    public class Category
    {
        [Key]
        public int CategoryID { get; set; }
        public required string Name { get; set; }
        public required string Description { get; set; }
        public DateTime CreatedAt { get; set; }
        public bool IsDeleted { get; set; }

        public ICollection<ArticleCategory> ArticleCategories { get; set; } = new List<ArticleCategory>();
    }
}

File: lib/category.dart
class Category {
  final int categoryID;
  final String name;

  Category({required this.categoryID, required this.name});

  factory Category.fromJson(Map<String, dynamic> json) {
    return Category(
      categoryID: json['categoryID'],
      name: json['name'],
    );
  }
}
